gpu/octree/src/octree.cpp

```diff
@@ -108,57 +108,73 @@ void pcl::gpu::Octree::radiusSearchHost(const PointType& center, float radius, s
 {
     if (!static_cast<OctreeImpl*>(impl)->host_octree.downloaded)
         internalDownload();
 
-    OctreeImpl::PointType q;
-    q.x = query.x;
-    q.y = query.y;
-    q.z = query.z;
-    
+    const OctreeImpl::PointType q = make_float4(query.x, query.y, query.z, NAN);
+
     static_cast<OctreeImpl*>(impl)->approxNearestSearchHost(q, out_index, sqr_dist);
 
 }
                         
 void pcl::gpu::Octree::radiusSearch(const Queries& queries, float radius, int max_results, NeighborIndices& results) const
 {
-    assert(queries.size() > 0);
-    results.create(static_cast<int> (queries.size()), max_results);
-    results.sizes.create(queries.size());
-    
+    pcl::gpu::Octree::ResultSqrDists sqr_distances;
     const OctreeImpl::Queries& q = (const OctreeImpl::Queries&)queries;
-    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radius, results);
-}
+    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radius, results, sqr_distances, false, max_results);}
 
 void pcl::gpu::Octree::radiusSearch(const Queries& queries, const Radiuses& radiuses, int max_results, NeighborIndices& results) const
 {
-    assert(queries.size() > 0);
-    assert(queries.size() == radiuses.size());
-    results.create(static_cast<int> (queries.size()), max_results);
-    results.sizes.create(queries.size());
-    
+    pcl::gpu::Octree::ResultSqrDists sqr_distances;
     const OctreeImpl::Queries& q = (const OctreeImpl::Queries&)queries;
-    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radiuses, results);
+    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radiuses, results, sqr_distances, false, max_results);
 }
 
 void pcl::gpu::Octree::radiusSearch(const Queries& queries, const Indices& indices, float radius, int max_results, NeighborIndices& results) const
 {
-    assert(queries.size() > 0 && indices.size() > 0);
-    results.create(static_cast<int> (indices.size()), max_results);
-    results.sizes.create(indices.size());
-    
+    pcl::gpu::Octree::ResultSqrDists sqr_distances;
+    const OctreeImpl::Queries& q = (const OctreeImpl::Queries&)queries;
+    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radius, results, sqr_distances, false, max_results);
+}
+
+void pcl::gpu::Octree::radiusSearch(const Queries& queries, const float radius, const int max_results, NeighborIndices& results, ResultSqrDists& sqr_distances) const
+{
+    const OctreeImpl::Queries& q = (const OctreeImpl::Queries&)queries;
+    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radius, results, sqr_distances, true, max_results);
+}
+
+void pcl::gpu::Octree::radiusSearch(const Queries& queries, const Radiuses& radiuses, const int max_results, NeighborIndices& results, ResultSqrDists& sqr_distances) const
+{
+    const OctreeImpl::Queries& q = (const OctreeImpl::Queries&)queries;
+    static_cast<OctreeImpl*>(impl)->radiusSearch(q, radiuses, results, sqr_distances, true, max_results);
+}
+
```

Comment:
This part of the file is still looking messy. I need to see what to do here. I'll let you know later.